python/python/bystro/covariance/_covariance_np.py

```diff
@@ -106,3 +95,146 @@ def _fill_prior_options(self, prior_options: dict[str, Any]) -> dict[str, Any]:
             "iw_params": {"pnu": 2, "sigma": 1.0},
         }
         return {**default_options, **prior_options}
+
+
+class LinearShrinkageCovariance(BaseCovariance):
+
+    def fit(self, X: NDArray[np.float_]) -> "LinearShrinkageCovariance":
+        """
+        This fits a covariance matrix using the linear shrinkage approach 
+        to combine the empirical covariance matrix with a structured 
+        estimator.
+
+        Parameters
+        ----------
+        X : np.array-like, (n_samples, n_covariates)
+            The centered data
+
```

Comment:
Let's also add a Raises section:

```py
Raises
------
ValueError:
     A value error will be raised if missing data is found in X ( np.isnan(X) evaluates to true ), or if X is not an NDArray
```

Reply:
This follows numpy conventions: https://numpydoc.readthedocs.io/en/latest/format.html#raises